akka-stream/src/main/scala/akka/stream/impl/StreamRefsMaster.scala

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2014-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.impl
+
+import akka.actor.{ ActorSystem, ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.annotation.InternalApi
+import akka.stream.StreamRefSettings
+
+/** INTERNAL API */
+@InternalApi
+private[stream] object StreamRefsMaster extends ExtensionId[StreamRefsMaster] with ExtensionIdProvider {
+
+  override def createExtension(system: ExtendedActorSystem): StreamRefsMaster =
+    new StreamRefsMaster(system)
+
+  override def lookup(): StreamRefsMaster.type = this
+
+  override def get(system: ActorSystem): StreamRefsMaster = super.get(system)
+}
+
+/** INTERNAL API */
+@InternalApi
+private[stream] final class StreamRefsMaster(system: ExtendedActorSystem) extends Extension {
+
+  val settings: StreamRefSettings = StreamRefSettings(system)
```

Comment:
Isn't this different from how we deal with other stream settings? Shouldn't it be part of `ActorMaterializerSettings` similar to `IOSettings`. One might want different settings for different materializations, e.g. different buffer capacity.

Reply:
Yeah, I was trying to insert it into the materializer settings, was a bit of a pain but probably the right thing to do... so will do.